### PR TITLE
Reinstated Bazel Windows local build

### PR DESCRIPTION
**Pull request by billfeng327**
reintroducing changes reverted by #18390 
A concern in #18390 was that it removed pollers for Linux CI builds, this PR sets up prerequisites for a Windows Bazel RBE build without affecting existing Linux builds.

### CODE REVIEW

**Comment by nicolasnoble** [comment]
Yeah, we just discussed this with Bill. We may need to get the Bazel team involved to figure out the right way to do this, if doable at all.

File: bazel/grpc_build_system.bzl [hunk 1]
```diff
@@ -161,7 +162,17 @@ def grpc_cc_test(name, srcs = [], deps = [], external_deps = [], args = [], data
         "exec_compatible_with": exec_compatible_with,
     }
     if uses_polling:
-        native.cc_test(testonly = True, tags = ["manual"], **args)
+        # Only run targets with pollers for non-MSVC
+        # Only run targets without pollers for MSVC
```
Comment (jtattermusch) [inline bazel/grpc_build_system.bzl]: the comment is out of date?

File: bazel/BUILD [hunk 2]
```diff
@@ -29,3 +29,16 @@ cc_grpc_library(
     proto_only = True,
     deps = [],
 )
+
```
Comment (jtattermusch) [inline bazel/BUILD]: looks like this is not being used anywhere? Let's only add once we actually have a use for this.

Comment (billfeng327) [reply bazel/BUILD]: removed

File: bazel/grpc_build_system.bzl (hunk 1 above)
Comment (billfeng327) [reply bazel/grpc_build_system.bzl]: changed

**Pull request merged**